src/core/title.js

```diff
@@ -69,6 +75,19 @@ function setDocumentTitle(conf, h1Elem) {
     documentTitle = `Preview of PR #${conf.prNumber}: ${documentTitle}`;
   }
 
+  if (conf.hasOwnProperty("level")) {
```

Comment:
I think we should move all this to a new plugin ("w3c/levels") that runs after core/title.